app/src/main/java/com/zulip/android/models/Message.java

```diff
@@ -419,6 +419,8 @@ public Void call() throws Exception {
                     for (Message m : messages) {
                         Person person = Person.getOrUpdate(app, m.getSenderEmail(), m.getSenderFullName(), m.getAvatarUrl());
                         m.setSender(person);
+                        Stream stream = Stream.getByName(app, m.getRecipients());
+                        m.setStream(stream);
```

Comment:
Apologies, I totally missed that, after giving it some thought I think setting streams to null for private messages can be a potential solution.